### PR TITLE
Preact retrocompatibility

### PR DESCRIPTION
**Pull request by CPatchane**
* Use portal from React only if `USE_PREACT` environment variable found
* Handle UNSAFE_ React API

### CODE REVIEW

File: react/Modal/index.jsx [hunk 1]
```diff
@@ -40,7 +40,13 @@ class ModalContent extends Component {
...
   }
 
-  UNSAFE_componentWillUpdate(nextProps) {
+  // for preact
+  // eslint-disable-next-line react/no-deprecated
+  componentWillUpdate() {
+    this.UNSAFE_componentWillReceiveProps()
+  }
+
+  UNSAFE_componentWillUpdate = nextProps => {
```
Comment (ptbrowne) [inline react/Modal/index.jsx]: UNSAFE_componentWillReceiveProps no ?

File: react/I18n/index.jsx [hunk 2]
```diff
@@ -35,7 +35,13 @@ export class I18n extends Component {
     }
   }
 
-  UNSAFE_componentWillReceiveProps(newProps) {
+  // for preact
+  // eslint-disable-next-line react/no-deprecated
+  componentWillUpdate() {
```
Comment (Crash--) [inline react/I18n/index.jsx]: componentWillReceiveProps

File: react/Modal/index.jsx [hunk 3]
```diff
@@ -40,7 +40,13 @@ class ModalContent extends Component {
     }
   }
 
-  UNSAFE_componentWillUpdate(nextProps) {
+  // for preact
+  // eslint-disable-next-line react/no-deprecated
+  componentWillUpdate() {
```
Comment (Crash--) [inline react/Modal/index.jsx]: omponentWillReceiveProps

File: react/Modal/index.jsx [hunk 4]
```diff
@@ -40,7 +40,13 @@ class ModalContent extends Component {
...
+    this.UNSAFE_componentWillReceiveProps()
```
Comment (Crash--) [inline react/Modal/index.jsx]: willUpdate

**Comment by CPatchane** [comment]
Sorry for these copy/paste typos, it's fixed, thantks @ptbrowne and @Crash-- 👍

**Comment by Crash--** [comment]
Looks good! Thx :)

**Comment by ptbrowne** [comment]
LGTM, thanks !

**Comment by Crash--** [comment]
🍾

File: react/Portal/index.jsx [hunk 5]
```diff
@@ -1,6 +1,12 @@
-import ReactDOM from 'react-dom'
-const Portal = ({ into, children }) => {
-  const targetElement = document.querySelector(into)
-  return ReactDOM.createPortal(children, targetElement)
+let Portal
+if (process.env.USE_REACT) {
```
Comment (enguerran) [inline react/Portal/index.jsx]: I know it is a breaking change and #594 should have said so. But, there is a very weird behavior here: we use React for everything by default, but we need to use a environment variable for Portal?

**Pull request merged**